mountpoint-s3/src/main.rs

```diff
@@ -12,92 +10,147 @@ use fuser::{MountOption, Session};
 
-    // log directories and files should not be accessible by other users
-    let mut file_options = OpenOptions::new();
-    file_options.mode(0o640).write(true).create(true);
+        /// Create the logging config from the RUST_LOG environment variable or the default config if
+        /// that variable is unset. We do this in a function because [EnvFilter] isn't [Clone] and we
+        /// need a second copy of it in the foreground case to replicate logs to stdout.
+        fn create_env_filter() -> EnvFilter {
+            EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info,awscrt=off,fuser=error"))
+        }
+        let env_filter = create_env_filter();
 
-    let mut dir_builder = DirBuilder::new();
-    dir_builder.recursive(true).mode(0o750);
+        // Don't create the files or subscribers if we'll never emit any logs
+        if env_filter.max_level_hint() == Some(LevelFilter::OFF) {
+            return Ok(());
+        }
 
-    let file = if let Some(path) = log_directory {
-        dir_builder.create(path).context("failed to create log folder")?;
-        file_options
-            .open(path.join(filename))
-            .context("failed to create log file")?
-    } else {
-        let default_log_directory = home::home_dir()
-            .ok_or(anyhow!("no home directory found"))?
-            .join(LOG_DIRECTORY);
-        dir_builder
-            .create(&default_log_directory)
-            .context("failed to create log folder")?;
-        file_options
-            .open(default_log_directory.join(filename))
-            .context("failed to create log file")?
-    };
+        RustLogAdapter::try_init().context("failed to initialize CRT logger")?;
 
-    let fmt_layer = tracing_subscriber::fmt::layer()
-        .with_ansi(false)
-        .with_writer(file)
-        .with_filter(env_filter);
-    let registry = tracing_subscriber::registry()
-        .with(fmt_layer)
-        .with(metrics_tracing_span_layer());
-
-    if is_foreground {
-        // Replicate logs to the console with the same filter applied
-        let fmt_layer_to_console = tracing_subscriber::fmt::layer()
-            .with_ansi(supports_color::on(supports_color::Stream::Stdout).is_some())
-            .with_filter(create_env_filter());
-        registry.with(fmt_layer_to_console).init();
-    } else {
-        registry.init();
-    }
+        let filename = OffsetDateTime::now_utc()
+            .format(LOG_FILE_NAME_FORMAT)
+            .context("couldn't format log file name")?;
 
-    Ok(())
+        // log directories and files should not be accessible by other users
+        let mut file_options = OpenOptions::new();
+        file_options.mode(0o640).write(true).create(true);
+
+        let mut dir_builder = DirBuilder::new();
+        dir_builder.recursive(true).mode(0o750);
+
+        let file = if let Some(path) = log_directory {
+            dir_builder.create(path).context("failed to create log folder")?;
```

Comment:
It is creating a log folder if it does not exist with the given permissions right? Should not it set the new permissions of the log directory if it already exist? Like if user has log directory created from previous build.

Reply:
This code hasn't changed (just indented differently), but yes, we don't change the permissions of a directory that already exists -- presumably the user wanted those permissions if they manually set them.